src/gui/attributetable/qgsattributetablefiltermodel.cpp

```diff
@@ -131,11 +131,29 @@ int QgsAttributeTableFilterModel::columnCount( const QModelIndex &parent ) const
 
 void QgsAttributeTableFilterModel::setAttributeTableConfig( const QgsAttributeTableConfig &config )
 {
+  QgsAttributeTableConfig oldConfig = mConfig;
   mConfig = config;
   mConfig.update( layer()->fields() );
 
-  QVector<int> newColumnMapping;
+  if ( oldConfig.columns().size() == mConfig.columns().size() )
+  {
+    bool resizeOnly = true;
+    int i = 0;
+    for ( i = 0; i < config.columns().size(); i++ )
+    {
+      if ( oldConfig.columns().at( i ).name != mConfig.columns().at( i ).name || oldConfig.columns().at( i ).type != mConfig.columns().at( i ).type )
```

Comment:
What about adding this directly to QgsAttributeTableConfig::ColumnConfig (e.g. "isResizeOnly( const QgsAttributeTableConfig::ColumnConfig& other )"), along with a unit test?

Keeping it next to QgsAttributeTableConfig::ColumnConfig::operator== will mean it's less likely to get overlooked if new members are ever added to ColumnConfig.

Reply:
Sure, can do.